packages/w3up-client/src/service.js

```diff
@@ -6,35 +8,68 @@ import { receiptsEndpoint } from '@storacha/upload-client'
 export const accessServiceURL = new URL('https://up.storacha.network')
 export const accessServicePrincipal = DID.parse('did:web:up.storacha.network')
 
-export const accessServiceConnection = client.connect({
-  id: accessServicePrincipal,
-  codec: CAR.outbound,
-  channel: HTTP.open({ url: accessServiceURL, method: 'POST' }),
-})
+const envName = isBrowser ? 'Browser' : isNode ? 'Node' : isBun ? 'Bun' : isDeno ? 'Deno' : isElectron ? 'Electron' : 'Unknown'
+export const defaultHeaders = {
+  'X-Client': `Storacha/1 (js; ${envName})`,
+}
+
+/**
+ * @param {object} [options]
+ * @param {Record<string, any>} [options.headers]
+ */
+export const accessServiceConnection = (options = {}) =>
+  client.connect({
+    id: accessServicePrincipal,
+    codec: CAR.outbound,
+    channel: HTTP.open({
+      url: accessServiceURL,
+      method: 'POST',
+      headers: { ...defaultHeaders, ...options.headers },
+    }),
+    ...options,
+  })
 
 export const uploadServiceURL = new URL('https://up.storacha.network')
 export const uploadServicePrincipal = DID.parse('did:web:up.storacha.network')
 
-export const uploadServiceConnection = client.connect({
-  id: uploadServicePrincipal,
-  codec: CAR.outbound,
-  channel: HTTP.open({ url: accessServiceURL, method: 'POST' }),
-})
+/**
+ * @param {object} [options]
+ * @param {Record<string, any>} [options.headers]
+ */
+export const uploadServiceConnection = (options = {}) =>
+  client.connect({
+    id: uploadServicePrincipal,
+    codec: CAR.outbound,
+    channel: HTTP.open({
+      url: accessServiceURL,
```

Comment:
@alanshaw I just noticed we are using `accessServiceURL` in the upload service connection. Is that correct?